File: node/translate.js
#!/usr/bin/env node
// AstraNL: minimal Node.js translate call
// Set ASTRANL_AGENT_KEY env var first.
const key = process.env.ASTRANL_AGENT_KEY;
if (!key) { console.error('set ASTRANL_AGENT_KEY env var'); process.exit(1); }

const resp = await fetch('https://astranl.com/capabilities/execute', {
  method: 'POST',
  headers: { 'X-Agent-Key': key, 'Content-Type': 'application/json' },
  body: JSON.stringify({
    intent: "translate to dutch: 'Hello, today is a great day'",
    capability_id: 'translate',
  }),
});
const data = await resp.json();
console.log('result:', data.result);
console.log('billing:', data.billing);

File: python/translate.py
#!/usr/bin/env python3
"""AstraNL: simplest possible translate call.

Set ASTRANL_AGENT_KEY environment variable first.
Get a key from: https://astranl.com/agents (or POST /api/agents/register)
"""
import os
import requests

KEY = os.environ.get("ASTRANL_AGENT_KEY")
if not KEY:
    raise SystemExit("set ASTRANL_AGENT_KEY env var")

resp = requests.post(
    "https://astranl.com/capabilities/execute",
    headers={"X-Agent-Key": KEY, "Content-Type": "application/json"},
    json={
        "intent": "translate to dutch: 'Hello world, today is a great day'",
        "capability_id": "translate",
    },
    timeout=30,
)
resp.raise_for_status()
data = resp.json()
print("result:", data["result"])
print("billing:", data.get("billing"))
print("wallet remaining (µEUR):", data.get("wallet_balance_micro_eur"))
